timeplus/client.go

```diff
@@ -404,30 +406,80 @@ func (s *TimeplusClient) QueryV2Stream(sql string, count int, bufferTime int) (r
 		res.Body.Close()
 	}()
 
-	observable := rxgo.FromChannel(ch, rxgo.WithPublishStrategy())
+	observable := rxgo.FromChannel(ch)
 	return observable, nil
 }
 
-func (e *ServerSideEvent) GetQuery() (*QueryEvent, error) {
-	if e.EventType != "query" {
+func (s *TimeplusClient) QueryStream(sql string, batchCount int, batchBufferTime int) (rxgo.Observable, *QueryInfo, error) {
+	stream, err := s.queryStreamV2(sql, batchCount, batchBufferTime)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	headerEvent := stream.Take(1)
+	contentStream := stream.Skip(1)
+	var header *serverSentEvent
+	sub := headerEvent.ForEach(func(v interface{}) {
+		event := v.(*serverSentEvent)
+		fmt.Printf("got header event %v\n", event)
+		header = event
+	}, func(err error) {
+		fmt.Printf("failed to query %s", err)
+	}, func() {
+
+	})
+
+	<-sub
+
+	dataStream := contentStream.Filter(func(v interface{}) bool {
+		// Filter operation
+		event := v.(*serverSentEvent)
+		return event.eventType == "data"
+	}).Map(func(_ context.Context, v interface{}) (interface{}, error) {
+		// Filter operation
+		event := v.(*serverSentEvent)
+		if data, err := event.GetData(); err != nil {
+			return nil, err
+		} else {
+			return data, nil
+		}
+	})
+
+	queryEvent, err := header.GetQuery()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var queryInfo QueryInfo
+	err = mapstructure.Decode(queryEvent, &queryInfo)
```

Comment:
I have mixed feeling on this. On one hand, using a struct as result can make it easier for the SDK user. On the other hand, we have to upgrade SDK every time we introduce changes to `query` object.

For now, as we don't have enough resource to properly maintain SDKs. IMO we should simply return queryInfo as `map[string]any` so that the user doesn't have to upgrade SDK regularly

Reply:
as some of the tools already use this API, I would like to keep minimal change for now, we can make a public one with map[string]any.